platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/nexus/LatestCompleteStateNexus.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.nexus;
+
+import static com.swirlds.common.metrics.Metrics.PLATFORM_CATEGORY;
+
+import com.swirlds.common.config.StateConfig;
+import com.swirlds.common.metrics.Metrics;
+import com.swirlds.common.metrics.RunningAverageMetric;
+import com.swirlds.platform.state.signed.ReservedSignedState;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+
+/**
+ * A nexus that holds the latest complete signed state.
+ */
+public class LatestCompleteStateNexus extends SignedStateNexus {
+    private static final RunningAverageMetric.Config AVG_ROUND_SUPERMAJORITY_CONFIG = new RunningAverageMetric.Config(
+                    PLATFORM_CATEGORY, "roundSup")
+            .withDescription("latest round with state signed by a supermajority")
+            .withUnit("round");
+
+    private final StateConfig stateConfig;
+
+    /**
+     * Create a new nexus that holds the latest complete signed state with no metrics.
+     *
+     * @param stateConfig the state configuration
+     */
+    public LatestCompleteStateNexus(@NonNull final StateConfig stateConfig) {
```

Comment:
Is the purpose of this constructor just for tests? If so, why not just mock the metrics object for the tests?

Reply:
good point, fixed